pkg/auth/auth.go

```diff
@@ -160,5 +168,17 @@ func normalizeHostname(host string) string {
 	if strings.HasSuffix(hostname, "."+localhost) {
 		return localhost
 	}
+	if before, found := cutSuffix(hostname, "."+tenancyHost); found {
+		idx := strings.LastIndex(before, ".")
+		return fmt.Sprintf("%s.%s", before[idx+1:], tenancyHost)
+	}
 	return hostname
 }
+
+// Backport strings.CutSuffix from Go 1.20.
```

Comment:
Strictly speaking we shouldn't need this backport anymore but I'd rather change `cli/cli` and this at the same time, and I don't want to change `cli/cli` right now.